source/common/io/io_uring_worker.h

```diff
@@ -0,0 +1,486 @@
+#pragma once
+
+#include "io_uring.h"
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/logger.h"
+
+#include "source/common/io/io_uring.h"
+#include "source/common/io/io_uring_impl.h"
+
+namespace Envoy {
+namespace Io {
+
+class IoUringAcceptSocket : public IoUringSocket, protected Logger::Loggable<Logger::Id::io> {
+public:
+  IoUringAcceptSocket(os_fd_t fd, IoUringHandler& io_uring_handler, IoUringWorker& parent) :
+    fd_(fd), io_uring_handler_(io_uring_handler), parent_(parent) {}
+
+  // IoUringSocket
+  os_fd_t fd() const override { return fd_; }
+  void start() override;
+  void close() override;
+  void enable() override;
+  void disable() override;
+
+  void onAccept(int32_t result) override;
+  void onClose(int32_t result) override;
+  void onCancel(int32_t result) override;
+private:
+  os_fd_t fd_;
+
+  IoUringHandler& io_uring_handler_;
+  IoUringWorker& parent_;
+
+  sockaddr_storage remote_addr_;
+  socklen_t remote_addr_len_{sizeof(remote_addr_)};
+  os_fd_t connection_fd_{INVALID_SOCKET};
+
+  Request* accept_req_{nullptr};
+  Request* cancel_req_{nullptr};
+  Request* close_req_{nullptr};
+
+  bool is_disabled_{false};
+  bool is_pending_accept_{false};
+  bool is_closing_{false};
```

Comment:
I've seen elsewhere in the codebase enums for these state transitions, it might make sense here instead of accept/is closing

Reply:
+1, that is what I'm thinking too, will do in the next iteration